File: src/main/java/ms/domwillia/jvmemory/monitor/LoadPrinter.java
package ms.domwillia.jvmemory.monitor;

public class LoadPrinter {

	private static void loadPrint(String type, int var) {
		System.out.printf("%s > load %s local var %d\n", StackTracker.getHead(), type, var);
	}
	public static void booleanDo(int var) {
		loadPrint("bool", var);
	}

	public static void charDo(int var) {
		loadPrint("char", var);
	}

	public static void byteDo(int var) {
		loadPrint("byte", var);
	}

	public static void shortDo(int var) {
		loadPrint("short", var);
	}

	public static void intDo(int var) {
		loadPrint("int", var);
	}

	public static void floatDo(int var) {
		loadPrint("float", var);
	}

	public static void longDo(int var) {
		loadPrint("long", var);
	}

	public static void doubleDo(int var) {
		loadPrint("double", var);
	}

	public static void objectDo(int var) {
		loadPrint("object", var);
	}
}

File: src/main/java/ms/domwillia/jvmemory/monitor/printer/LoadPrinter.kt
package ms.domwillia.jvmemory.monitor.printer

import ms.domwillia.jvmemory.monitor.StackTracker

class LoadPrinter(private val stackTracker: StackTracker) {

    private fun loadPrint(type: String, index: Int) {
        println("${stackTracker.head} > load $type local var $index")
    }

    fun booleanDo(index: Int) {
        loadPrint("bool", index)
    }

    fun charDo(index: Int) {
        loadPrint("char", index)
    }

    fun byteDo(index: Int) {
        loadPrint("byte", index)
    }

    fun shortDo(index: Int) {
        loadPrint("short", index)
    }

    fun intDo(index: Int) {
        loadPrint("int", index)
    }

    fun floatDo(index: Int) {
        loadPrint("float", index)
    }

    fun longDo(index: Int) {
        loadPrint("long", index)
    }

    fun doubleDo(index: Int) {
        loadPrint("double", index)
    }

    fun objectDo(index: Int) {
        loadPrint("object", index)
    }
}